src/meta/meta_ckpt.c

```diff
@@ -908,17 +927,13 @@ __ckpt_load(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *k, WT_CONFIG_ITEM *v, WT_C
 
     WT_RET(__wt_config_subgets(session, v, "order", &a));
     if (a.len == 0)
-        goto format;
+        WT_RET_MSG(session, WT_ERROR, "corrupted order value in checkpoint config");
     ckpt->order = a.val;
 
     WT_RET(__wt_config_subgets(session, v, "time", &a));
-    if (a.len == 0 || a.len > sizeof(timebuf) - 1)
-        goto format;
-    memcpy(timebuf, a.str, a.len);
-    timebuf[a.len] = '\0';
-    /* NOLINTNEXTLINE(cert-err34-c) */
-    if (sscanf(timebuf, "%" SCNu64, &ckpt->sec) != 1)
-        goto format;
+    ret = __ckpt_parse_time(session, &a, &ckpt->sec);
+    if (ret != 0)
+        WT_RET_MSG(session, WT_ERROR, "corrupted time value in checkpoint config");
```

Comment:
Should we pass in `ret` here in case `__ckpt_parse_time` evolves and may return other error codes? This comment applies wherever we call `WT_RET_MSG` after `__ckpt_parse_time` returns a non-zero value.

Reply:
It probably looks like this code knows that the only error that `ckpt_parse_time()` can return is `WT_ERROR` and therefore that's the error we return in `WT_RET_MSG`. But that's not what happened here. 

I'm simply matching the error behavior this function had before I changed it. All of the cases where we fail to parse the value of a required field in the checkpoint config used to land at the `format:` label, where we returned `WT_ERROR`. I'm just returning that same error explicitly from each of those sites.

IMO, I think it is safer to just return the same errors that callers already expect. I'm sure somebody thought about the return values here when this code was written (a long time ago), but my guess is that these error paths only come up in development -- the only likely way for the checkpoint config to be wrong is if WT wrote it incorrectly. So as long as we flag the problem (i.e., with the error message) it's probably fine.